File: app/src/main/java/com/eccentricyan/pohe/common/migration/Migration.java
package com.eccentricyan.pohe.common.migration;

import io.realm.DynamicRealm;
import io.realm.RealmMigration;

/**
 * Created by shiyanhui on 2017/04/15.
 */

public class Migration implements RealmMigration {
    @Override
    public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {

        // For now anyone migration is needed!
    }

    // https://github.com/realm/realm-java/issues/1919#issuecomment-165375852
    @Override
    public boolean equals(Object o) {
        return o instanceof Migration;
    }
}

File: app/src/main/java/com/eccentricyan/pohe/common/migration/Migration.kt
package com.eccentricyan.pohe.common.migration

import io.realm.DynamicRealm
import io.realm.RealmMigration

/**
 * Created by shiyanhui on 2017/04/15.
 */

class Migration : RealmMigration {
    override fun migrate(realm: DynamicRealm, oldVersion: Long, newVersion: Long) {

        // For now anyone migration is needed!
    }

    // https://github.com/realm/realm-java/issues/1919#issuecomment-165375852
    override fun equals(o: Any?): Boolean {
        return o is Migration
    }
}

